src/lib/rng/hmac_drbg/hmac_drbg.h

```diff
@@ -8,26 +8,65 @@
 #ifndef BOTAN_HMAC_DRBG_H__
 #define BOTAN_HMAC_DRBG_H__
 
-#include <botan/rng.h>
+#include <botan/stateful_rng.h>
 #include <botan/mac.h>
 
 namespace Botan {
 
+class Entropy_Sources;
+
 /**
 * HMAC_DRBG from NIST SP800-90A
 */
 class BOTAN_DLL HMAC_DRBG final : public Stateful_RNG
    {
    public:
       /**
-      * Initialize an HMAC_DRBG instance with the given hash function
+      * Initialize an HMAC_DRBG instance with the given MAC as PRF (normally HMAC)
+      *
+      * Automatic reseeding is disabled completely, as it as no access to
+      * any source for seed material.
+      *
+      * If a fork is detected, the RNG will be unable to reseed itself
+      * in response. In this case, an exception will be thrown rather
+      * than generating duplicated output.
+      */
+      HMAC_DRBG(std::unique_ptr<MessageAuthenticationCode> prf);
```

Comment:
NIST SP800-90A says "HMAC_DRBG uses multiple occurrences of an **approved** keyed hash function, which is based on an **approved** hash function.", sec. 10.1.2. The only NIST-**approved** keyed hash function currently is HMAC, but other organizations may permit other keyed hash functions as well.